### ISSUE TITLE
How to catch a Kratos::exception with python ?

### ISSUE THREAD

**Issue by josep-m-carbonell** [KratosMultiphysics/Kratos#2833]
Are Kratos::exeptions translated to python by pybind?  (https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html)
If they are not, can be translated?
Is it possible to catch them from python in order to write a custom python except clause?

**Comment by loumalouomega** [comment]
I don't see any export here https://github.com/KratosMultiphysics/Kratos/tree/master/kratos/python

**Comment by loumalouomega** [comment]
I really like the idea, but I don't have the experience in pybind to do it, sorry.

**Comment by josep-m-carbonell** [comment]
@roigcarlo  is going to open a new issue in order to make it work properly. Don't worry.

**Comment by roigcarlo** [comment]
On it, still making a couple more tests since I want to be sure about which point of the macro is giving problems

**Comment by roigcarlo** [comment]
I will reopen. So after debugging the issue with @josep-m-carbonell we discovered that it fails because the exception is being raised in a parallel region.

Honestly I have no idea how to elegantly handle exceptions in parallel. After a quick search the overall opinion seems to be that exceptions inside a parallel region need to be handled inside the parallel region that caused them, which is quite problematic for us as it means we have to change pretty much every parallel loop inside Kratos.

If someone in the @KratosMultiphysics/technical-committee knows how to deal with this it will be highly appreciated. I will add this to the TODO list.

**Comment by jcotela** [comment]
@roigcarlo just to be clear, do we have the capability to catch Kratos exceptions in python in general? Or are some changes needed? If so, I'd do the PR anyway, since I believe this is a worthwhile feature.

Regarding the parallel region error, if you tell me how to reproduce the error, I'd take a look too.

**Comment by roigcarlo** [comment]
Basically, it cannot be done because openmp does not handle the exceptions properly. There is nothing we can do about it, I will update the wiki so it is clearly reflected.

**Comment by josep-m-carbonell** [comment]
I hope that this way of solving an issue is not the best way. I think exceptions can be captured modifying the loops. And if this option is not good enough maybe there is another way to keep the execution alive without throwing errors. I understand that to solve a problem a general interest is needed, I guess that in this case the interest is not shared by the technical committee.

**Comment by maceligueta** [comment]
Dear @josep-m-carbonell, as you can see in the previous messages of this very issue, a lot of interest has been put in the problem by the members of the Committee. If we have not reached a solution after all the time dedicated to it, that does not mean we did not want to.
Of course, we are open to specific solutions proposed by anyone who is capable to bring light to this. If that was the case, this issue can be re-opened for reevaluation.
Do not hesitate to paste your piece of code for that.
